src/GCodePathConfig.h

```diff
@@ -13,97 +13,73 @@ namespace cura
  */
 class GCodePathConfig
 {
-    friend class GCodePlannerTest;
+    friend class LayerPlanTest;
```

Comment:
Again?! Still!

The friendship is the other way around than you think.
The test case is meddling with private data because the test case isn't using the normal flow of the engine...